Start script improvements

jsoref opened the pull request:
* it's possible to end up w/ "Windows" line endings in `ongdb.conf` and when that happens running the unix `ongdb` release script with that file results in a very unhappy system. Tolerating those line endings seems reasonable, as users generally expect conf files to be portable and not to need to worry about line endings.
* one should be able to run `ongdb version` without java preallocated gb's of memory for `ongdb` -- we have containers that run with enough space for one copy of `ongdb` but definitely not enough space for two copies of `ongdb`, but it should be possible to run `ongdb version` as that doesn't require the preallocation defined in `ongdb.conf` -- Simplify the start script to do minimal work when getting version (skipping `setup_java_opts`, most of `setup_arbiter_options`).

jsoref committed:
Ignore line endings in config file

jsoref committed:
Ignore line endings in config file

jsoref committed:
Simplify version handling

ongdb does not need special memory information when running version

jsoref committed:
Simplify version handling

ongdb does not need special memory information when running version

bradnussbaum commented:
These changes seem reasonable to me. I'm good with both of them. Build is failing during testing of server assembly. I think we discussed elsewhere that test expectations need updated following the changes to the version command.

bradnussbaum committed:
Harden start-script fixes in m4 sources and tests

Sync CRLF stripping and lightweight version handling into ongdb.m4 /
ongdb-shared.m4, drop heap opts from the java -version probe, and add
Sharness coverage so ongdb version is verified without a full package.
